hedgehog/src/Hedgehog/Internal/Property.hs

```diff
@@ -449,6 +469,13 @@ liftExceptT :: (Monad m, Show x, HasCallStack) => ExceptT x m a -> Test m a
 liftExceptT m =
   withFrozenCallStack liftEither =<< lift (runExceptT m)
 
+-- | Fails the test if the 'ExceptT' is 'Left', otherwise returns the value in
+--   the 'Right'.
+--
+withExceptT :: (Monad m, Show x, HasCallStack) => Test (ExceptT x m) a -> Test m a
+withExceptT m =
+  withFrozenCallStack liftEither =<< runExceptT (distribute m)
```

Comment:
I'm going to go with this for now, if we decide to pick a different name it will be a breaking change anyway because of `withResourceT` so I don't see a reason to hold this up.